### ISSUE TITLE
Dynamic module imports not getting resolved

### ISSUE THREAD

**Issue by timoruppell** [cypress-io/cypress#9689]
This is probably not a bug. I suspect a mis-configuration on my part or a technical limitation of how unit testing works in cypress.

#### Summary

I am trying to set up component testing for a quite extensive Vue app and am running into problems with the `@` alias not getting resolved leading to errors like this one:

`Error: Cannot find module '@/components /<path_to_child_component>'`

when the component to be tested contains a child component that is lazy loaded from that path.

#### Setup

I am using [lazy loading](https://router.vuejs.org/guide/advanced/lazy-loading.html) to import child components via this function:

```javascript
export function lazyImport (component) {
    return () => import(/* webpackChunkName: "view-[request]" */ `@/components/${component}`);
}
``` 

I also use lazy loading for i18n as per their [instructions](https://kazupon.github.io/vue-i18n/guide/lazy-loading.html).

Cypress plugin setup (vanilla + splicing in webpack options):
```javascript
const preprocessor = require("@cypress/vue/dist/plugins/webpack");
const webpackOptions = require("../../node_modules/@vue/cli-service/webpack.config.js");

module.exports = (on, config) => {
    const _config = {
        ...config,
        webpackOptions,
    };
    preprocessor(on, _config);
    // IMPORTANT return the config object
    return _config;
};
```

Relevant part of the `webpackOptions` used in the plugin setup above:
```javascript
{
  mode: 'development',
  resolve: {
    alias: {
      '@': '/Users/ruppell/Code/mapita/maptionnaire2/src/client/src',
      'vue$': 'vue/dist/vue.esm.js'
    },
  plugins: [
    VueLoaderPlugin {},
  ],
    ...
}
```

Cypress config:
```
{
  "baseUrl": "http://localhost:8080",
  "pluginsFile": "tests/plugins/index.js",
  "supportFile": "tests/support/index.js",
  "fixturesFolder": "tests/fixtures",
  "integrationFolder": "tests/e2e",
  "testFiles": "**/*.spec.js*",
  "experimentalComponentTesting": true,
  "componentFolder": "tests/components",
  "nodeVersion": "system"
}
```

My dev environment:
```
macOS: 10.15.7 (19H2)
Node: v12.18.4
Chrome: Version 86.0.4240.80 (Official Build) (x86_64)
```

My dev dependencies:
```
"@cypress/code-coverage": "3.8.1",
"@cypress/vue": "1.0.0-alpha.4",
"@vue/babel-preset-app": "4.5.7",
"@vue/cli-plugin-babel": "4.5.7",
"@vue/cli-plugin-e2e-cypress": "4.5.7",
"@vue/cli-plugin-eslint": "4.5.7",
"@vue/cli-service": "4.5.7",
"babel-eslint": "10.0.3",
"babel-plugin-istanbul": "6.0.0",
"cypress": "5.3.0",
"eslint": "5.16.0",
"eslint-plugin-vue": "6.1.2",
"node-sass": "4.14.1",
"sass-loader": "10.0.3",
"string-replace-loader": "2.2.0",
"vue-cli-plugin-cypress-experimental": "1.2.0",
"vue-template-compiler": "2.6.11",
"webpack": "4.41.6",
"webpack-bundle-analyzer": "3.6.1"
```

#### Symptoms

The `@` alias resolution works in the test spec. E.g. importing the component  to be tested and other dependencies works just fine:
```javascript
import {  mount } from "@cypress/vue";

import AssetCard from "@/components/base-elements/AssetCard.vue";

import Buefy from "buefy";
import icons from "@/icons";
...
```

But the AssetCard.vue component tries to lazy load it's child components and for those imports the `@` alias resolution does not seem to work. 

#### Possible mitigations 

I have verified that the proper alias resolution options get passed to webpack, I have tried separately invoking `on("file:preprocessor", webpack(webpackOptions))` (possibly not relevant and also shouldn't be necessary as it is being invoked by cypress later). 

I can make use of the `components` option of the `mount()` function to avoid all of the errors relating to child components. However, this involves loading all the components in the chain of child components which makes the test highly coupled to irrelevant implementation details.

I can change the lazy loading functions to use absolute paths. This is not ideal as it requires additional logic to make test and production server builds work nicely.

#### Question

Is there just something that I am doing wrong here? Is there just a technical problem to using dynamic imports in this kind of setup? Any help in figuring this out is greatly appreciated!

**Comment by laland** [comment]
@timoruppell i have exactly same problem with dynamic imports, but i have a kinda workaround =). The root of the problem is that cypres won't serve static assets, produced by webpack build. Only the main entry bundle would be served. So i created a webpack plugin:
```typescript
// cypres/plugins/fix-bundle-related-files-plugin.ts
import path from 'path';
import { Compiler, Plugin } from 'webpack';
import fse from 'fs-extra';

export class FixBundleRelatedFilesPLugin implements Plugin {
  apply(compiler: Compiler): void {
    const distPath = path.resolve(__dirname, '../../dist');

    compiler.hooks.afterEnvironment.tap('FixBundleRelatedFilesPLugin', () => {
      const srcFolder = 'bundles/src';
      const outputOptions = compiler.options.output;
      if (!(new RegExp(srcFolder)).test(outputOptions.path)) {
        return;
      }

      const [bundleDistAbsoltePath, componentPublicPath] = outputOptions.path.split(srcFolder);

      outputOptions.publicPath = `/${componentPublicPath}/`.replace('//', '/');

      const bundleDistFolder = bundleDistAbsoltePath + srcFolder;

      console.log('COPY START', bundleDistFolder, distPath);
      fse.remove(`${distPath}/bundles`);
      fse.copySync(bundleDistFolder, distPath, { overwrite: true });
      console.log('COPY END', bundleDistFolder, distPath);
    });
  }
}
```

and then used it in `cypress/plugins/index.ts`, like this:
```typescript
import preprocessor from '@cypress/webpack-preprocessor';
import * as Cypress from 'cypress';
import { Configuration } from 'webpack';
import { getWebpackAliases } from '../../config/webpack/aliases';
import { cssLoader } from '../../config/webpack/css-loader';
import { fontsLoader } from '../../config/webpack/fonts-loader';
import { generatePurchaseRules } from '../../config/webpack/generate-purchase-rules';
import { tsLoader } from '../../config/webpack/ts-loader';
import { vueLoader } from '../../config/webpack/vue-loader';
import { vuePlugins } from '../../config/webpack/vue-plugins';
import { FixBundleRelatedFilesPLugin } from './fix-bundle-related-files-plugin';
import PluginConfig = Cypress.PluginConfig;

const webpackOptions = <Configuration>{
  module: {
    rules: [
      fontsLoader,
      cssLoader,
      ...generatePurchaseRules(),
      ...preprocessor.defaultOptions.webpackOptions.module.rules,
      vueLoader,
      tsLoader
    ]
  },
  resolve: {
    alias: getWebpackAliases(),
    extensions: ['.js', '.ts', '.vue']
  },
  plugins: [
    ...vuePlugins,
    new FixBundleRelatedFilesPLugin()
  ]
};

export default <PluginConfig>((on, config) => {
  on('file:preprocessor', preprocessor({
    ...config,
    webpackOptions
  }));
});
```

and it worked for me, dynamic imports, fonts and image loading - all works fine with that.

**Comment by lmiller1990** [comment]
Hi everyone!

We just released Cypress 7.0, which comes with a completely revamped Component Testing experience. 

Updated Component Testing intro: https://docs.cypress.io/guides/component-testing/introduction
Changelog: https://docs.cypress.io/guides/references/changelog
Migration guide: https://docs.cypress.io/guides/references/migration-guide#Component-Testing

[Read the intro blog post](https://www.cypress.io/blog/2021/04/06/introducing-the-cypress-component-test-runner/) or the [getting start with Vue 2/3 blogpost](https://www.cypress.io/blog/2021/04/06/getting-start-with-cypress-component-testing-vue-2-3/).

The new dev server architecture seems to solve this problem. I tried it out [see lazy load with `@` here](https://github.com/lmiller1990/vue-cypress-template/blob/dbb40988d88d4265d77161561c092beceb7ca36f/src/components/ClickToLazyLoad.vue#L18) and spec file [here](https://github.com/lmiller1990/vue-cypress-template/blob/dbb40988d88d4265d77161561c092beceb7ca36f/src/components/ClickToLazyLoad.spec.ts).

Please try upgrading (see migration guide). If you still have problems, post here and we can figure out what's going on.
